src/core/hoc/StylesInjection.tsx

```diff
@@ -2,16 +2,19 @@ import * as React from 'react';
 import { StyleSheetManager } from 'styled-components';
 import createCache from '@emotion/cache';
 import { CacheProvider } from '@emotion/react';
+import { customAlphabet } from 'nanoid';
 
-interface Props extends React.HTMLProps<HTMLDivElement> {
-    id: string;
+interface Props extends Pick<React.HTMLProps<HTMLDivElement>, 'children'> {
     injectionPoint?: HTMLElement | undefined;
 }
 
 export const StylesInjection = (props: Props) => {
 
+  // @emotion/cache only accept alpha characters
+  const generateUniqueAlphaCharacter = customAlphabet('abcdefghijklmnopqrstuvwxyz', 16);
+
   const emotionCache = createCache({
-    key: props.id,
+    key: generateUniqueAlphaCharacter(),
```

Comment:
One of my previous comments seems to have disappeared. 
Could you generate the ID only if it is not provided in props? (To keep a way to specify it if needed)
`key: props.id || generateUniqueAlphaCharacter()`

Reply:
I resolved it so it's hidden.

But I don't think we should allow an id to be passed. There is no point really for users to provide their own id, it will only give them a headache and risk for possible problems if they don't provide a unique id.

Reply:
Ok Thanks